### PR TITLE
Trost Aug 17 Fixes

### PR DESCRIPTION
**Pull request by tsortwehttam**
OK to merge.

Short review.

Summary of changes:

- Pretty decent `hover`/`unhover` events
- Various fixes (see commit messages)

### CODE REVIEW

**Comment by tsortwehttam** [comment]
RECHECK!

File: packages/haiku-serialization/src/bll/Row.js [hunk 1]
```diff
@@ -282,6 +282,13 @@ class Row extends BaseModel {
     this.destroy()
   }
 
+  visit (visitor) {
+    visitor(this)
+    this.children.forEach((child) => {
+      child.visit(visitor)
+    })
+  }
+
```
Comment (stristr) [inline packages/haiku-serialization/src/bll/Row.js:291]: Oh good, another one :trollface:!

**Comment by stristr** [comment]
Code changes in `src/` directories LGTM.

File: packages/haiku-serialization/src/bll/Row.js (hunk 1 above)
Comment (tsortwehttam) [reply packages/haiku-serialization/src/bll/Row.js:291]: :P For now I just tell myself that, in a world of trees, writing a `visitor` is like writing a `for` loop.

(In all seriousness, one day I will consolidate these. )

**Pull request merged**